core/compiler/src/main/scala/com/fortysevendeg/exercises/compiler/comments.scala

```diff
@@ -0,0 +1,341 @@
+/*
+ * scala-exercises-exercise-compiler
+ * Copyright (C) 2015-2016 47 Degrees, LLC. <http://www.47deg.com>
+ */
+
+package com.fortysevendeg.exercises
+package compiler
+
+import scala.language.higherKinds
+
+import scala.tools.nsc._
+import scala.tools.nsc.doc.base.CommentFactoryBase
+import scala.tools.nsc.doc.base.MemberLookupBase
+import scala.tools.nsc.doc.base.LinkTo
+import scala.tools.nsc.doc.base.LinkToExternal
+import scala.tools.nsc.doc.base.comment._
+
+import scala.xml.NodeSeq
+import scala.xml.Xhtml
+
+import scalariform.formatter.{ ScalaFormatter }
+
+import cats.{ Id, Eq, Functor }
+import cats.data.Xor
+
+/** Facade for the different layers of comment processing. */
+object Comments {
+  import CommentZed._
+  import CommentParsing.ParseK
+
+  /** Type capturing the types for the name, description,
+    * and explanation fields.
+    */
+  type Mode = {
+    /** Name field type */
+    type N[A]
+    /** Description field type */
+    type D[A]
+    /** Explanation field type */
+    type E[A]
+  }
+
+  object Mode {
+    type Aux[N0[_], D0[_], E0[_]] = Mode {
+      type N[A] = N0[A]
+      type D[A] = D0[A]
+      type E[A] = E0[A]
+    }
+
+    /** Library comments. Required name. Required description.
+      * Require no description.
```

Comment:
I think you mean "explanation" instead of "description", could be? Also in the next type's comment.